src/parser.cpp

```diff
@@ -106,6 +120,273 @@ Buffer Usage
 */
 //-----------------------------------------------
 
+namespace {
+
+struct discontiguous_iterator
```

Comment:
can't say I'm a fan but... let's see where this goes

Reply:
It simplifies code in certain places, like parsing a last chunk header that's just `0000000000000000000000000\r\n\r\n` or something like this.

We can error out in cases like this but eh, it was kind of easy to write this helper.